src/main/java/io/appform/dropwizard/discovery/bundle/ServiceDiscoveryBundle.java

```diff
@@ -185,11 +187,16 @@ protected int getPort(T configuration) {
     }
 
     protected String getHost() throws UnknownHostException {
-        val host = serviceDiscoveryConfiguration.getPublishedHost();
+        String host = serviceDiscoveryConfiguration.getPublishedHost();
         if (Strings.isNullOrEmpty(host) || host.equals(Constants.DEFAULT_HOST)) {
-            return InetAddress.getLocalHost()
+            host = InetAddress.getLocalHost()
                     .getCanonicalHostName();
         }
+        Preconditions.checkArgument(
+                !LOCALHOSTS.contains(host) ||
+                        LOCALHOSTS.stream().anyMatch(localhost -> serviceDiscoveryConfiguration.getZookeeper().contains(localhost)),
+                "Looks like publishedHost has been set to localhost/127.0.0.1 and zookeeper has not been set to localhost/127.0.0.1. This is wrong. \n" +
+                        "Set zookeeper host to localhost/127.0.0.1 in config in order to set publishedHost as localhost/127.0.0.1 for the running service");
```

Comment:
The check here will need to be much more involved for situations where someone has set some hostname for localhost for whatever reason.

Reply:
only allowing publishedHost as local address with zookeeper pointed to  local address. Please suggest more precondition checks

Reply:
Have put check for custom hostnames configured for localhost, 127.0.0.1, 127.0.1.1